File: Backend/src/types/email.ts
export interface EmailInput {
  email: {
    subject: string;
    sender: string;
    body: string;
  };
  labels: {
    name: string;
    description: string;
  }[];
}

export interface EmailOutput {
  label: string;
  important: boolean;
}

// 🔥 NEW (batch)
export interface BatchEmailInput {
  emails: {
    id: string;
    subject: string;
    sender: string;
    body: string;
  }[];
  labels: {
    name: string;
    tags: string[];
  }[];
}

export interface BatchEmailOutput {
  id: string;
  label: string;
  important: boolean;
}
export interface Email {
  id: string;
  subject: string;
  sender: string;
  snippet: string;
}

File: AI-Services/app/schemas/email.py
from pydantic import BaseModel
from typing import List

class Label(BaseModel):
    name: str
    description: str

class EmailData(BaseModel):
    subject: str
    sender: str
    body: str

class EmailInput(BaseModel):
    email: EmailData
    labels: List[Label]

class EmailOutput(BaseModel):
    label: str
    important: bool